src/main/java/edu/usf/cutr/gtfsrtvalidator/api/resource/GtfsRtFeed.java

```diff
@@ -221,4 +251,39 @@ public synchronized static ScheduledExecutorService startBackgroundTask(GtfsRtFe
             return runningTasks.get(rtFeedUrl);
         }
     }
+
+    public String getDateFormat(long lastOccurrence, int gtfsRtId) {
+        Session session = GTFSDB.InitSessionBeginTrans();
+        GtfsRtFeedModel gtfsRtFeed = (GtfsRtFeedModel) session.createQuery(" FROM GtfsRtFeedModel "
+                + "WHERE rtFeedID = " + gtfsRtId).uniqueResult();
+        GTFSDB.commitAndCloseSession(session);
+
+        GtfsDaoImpl gtfsFeed = GtfsFeed.GtfsDaoMap.get(gtfsRtFeed.getGtfsId());
+        agencyTimezone = gtfsFeed.getAllAgencies().iterator().next().getTimezone();
+
+        DateFormat todaytimeFormat = new SimpleDateFormat("hh:mm:ss a");
+        DateFormat dateTimeFormat = new SimpleDateFormat("yyyy-MM-dd hh:mm:ss a");
+        DateFormat todayDateFormat = new SimpleDateFormat("yyyy-MM-dd");
+
+        TimeZone timeZone = TimeZone.getTimeZone(agencyTimezone);
+        todaytimeFormat.setTimeZone(timeZone);
+        dateTimeFormat.setTimeZone(timeZone);
+        todayDateFormat.setTimeZone(timeZone);
+
+        String formattedTime;
+        String currentDate = todayDateFormat.format(new Date().getTime()); // converting to milliseconds
+        long fromStartOfDay = 0;
+        try {
+            fromStartOfDay = todayDateFormat.parse(currentDate).getTime() / 1000; // converting to seconds
```

Comment:
For all time conversions, please use the Java `TimeUnit` instead of manually dividing or multiplying.  For example, this should be `fromStartOfDay = TimeUnit.MILLISECONDS.toSeconds(todayDateFormat.parse(currentDate).getTime());`

This helps avoid errors in time conversion calculations.  See http://www.javaworld.com/article/2074114/core-java/the-highly-useful-java-timeunit-enum.html for details.